### PR TITLE
Use metaextract

### PR DESCRIPTION
**Pull request by toabctl**

### CODE REVIEW

File: python_requires [hunk 1]
```diff
@@ -18,32 +18,37 @@ import argparse
...
+def _requires_get_source(all_requires, pkg):
+    if pkg in all_requires['install']:
+        return 'install'
+    elif pkg in all_requires['extras']:
+        return 'extras'
+    elif pkg in all_requires['tests']:
+        return 'tests'
+    else:
+        return 'unknown'
+
```
Comment (aplanas) [inline python_requires]: Maybe?

```
for key in all_requires:
    if pkg in all_requires[key]:
        return key
return 'unknown'
```

File: python_requires [hunk 2]
```diff
@@ -18,32 +18,37 @@ import argparse
...
 
-    return file_content
+def _get_complete_requires(all_requires):
+    """get a dict with all requirements from 'install', 'extras' and 'tests'"""
+    complete_requires = dict()
+    complete_requires.update(all_requires['install'])
+    complete_requires.update(all_requires['extras'])
+    complete_requires.update(all_requires['tests'])
```
Comment (aplanas) [inline python_requires]: Not sure here, but this order will make more important 'tests' that 'install'. So 'tests' can overwrite 'install' and 'extras'. Maybe use the reverse order for updating?

File: python_requires [hunk 3]
```diff
@@ -18,32 +18,37 @@ import argparse
...
-def parse_update_spec_file(specfile, contents, run_requires):
+def parse_update_spec_file(specfile, contents, all_requires):
     found_requires = set()
-    all_requires = set()
+    # complete dict of requires ('install', 'extras' and 'tests')
+    complete_requires = _get_complete_requires(all_requires)
 
-    for d in run_requires:
-        all_requires.add(d)
+    for d in complete_requires.keys():
```
Comment (aplanas) [inline python_requires]: you can skip the `keys()` call: dicts are iterable by keys

File: python_requires [hunk 4]
```diff
@@ -61,27 +66,29 @@ def parse_update_spec_file(specfile, contents, run_requires):
                              contents, flags=re.MULTILINE | re.IGNORECASE):
             spec_requires.add("python-%s" % m.group(1))
 
-        if all_requires.difference(found_requires):
-            for d in all_requires.difference(found_requires):
-                print("ERROR: %s is missing in %s "
-                      "(source: %s)" % (d, specfile, run_requires[d][1]))
+        complete_requires_keys = set(complete_requires.keys())
```
Comment (aplanas) [inline python_requires]: again, you can skip the call to `keys()`

File: python_requires (hunk 1 above)
Comment (toabctl) [inline python_requires]: Hm. I don't expect something different than the 3 keys. So I would leave them explicit here

File: python_requires (hunk 2 above)
Comment (toabctl) [inline python_requires]: done

File: python_requires (hunk 3 above)
Comment (toabctl) [inline python_requires]: done

File: python_requires (hunk 4 above)
Comment (toabctl) [inline python_requires]: done

**Comment by toabctl** [comment]
@aplanas updated

**Comment by aplanas** [comment]
+1

**Pull request merged**